tuna/miopen/worker/fin_eval.py

```diff
@@ -300,14 +300,13 @@ def manage_queue(self):
   def check_env(self) -> bool:
     """Check the GPU on the machine matches the GPU specified in session table"""
     if super().check_env():
-      if self.dbt.session.arch != self.machine.arch:
-        raise ValueError(
-            f'session arch {self.dbt.session.arch} does not match machine arch\
-            {self.machine.arch}')
-      if self.dbt.session.num_cu != self.machine.num_cu:
-        raise ValueError(
-            f'session num_cu {self.dbt.session.num_cu} does not match machine num_cu\
-            {self.machine.num_cu}')
+      if self.dbt.session.arch != self.machine.arch or \
+              self.dbt.session.num_cu != self.machine.num_cu:
+        self.logger.error(
+            'Session arch/num_cu (%s/%s) does not match env arch/num_cu (%s/%s)',
+            self.dbt.session.arch, self.dbt.session.num_cu, self.machine.arch,
+            self.machine.num_cu)
+        return False
```

Comment:
@cderb  would this be better moved in the fin_class check_env function? that way both eval and compile can check?

Reply:
Compile is capable of running on other architectures.
Though applicability needs to be run on the same architecture because of CK.
So there is some complexity here.